include/ensmallen_bits/callbacks/progress_bar.hpp

```diff
@@ -132,15 +124,24 @@ class ProgressBar
    * @param objective Objective value of the current point.
    */
   template<typename OptimizerType, typename FunctionType, typename MatType>
-  void StepTaken(OptimizerType& /* optimizer */,
+  void StepTaken(OptimizerType& optimizer,
                  FunctionType& /* function */,
                  const MatType& /* coordinates */)
   {
     if (newEpoch)
     {
-      output << "Epoch " << epoch << "/" << epochs << "\n";
-      output.flush();
-      newEpoch = false;
+      if (!optimizer.MaxIterations())
+      {
+        output << "Epoch " << epoch << "\n";
+        output.flush();
+        newEpoch = false;
+      }
+      else
+      {
+        output << "Epoch " << epoch << "/" << epochs << "\n";
+        output.flush();
+        newEpoch = false;
+      }
```

Comment:
Here's a simpler way to write this; what do you think?

```suggestion
      output << "Epoch " << epoch;
      if (epochs > 0)
        output << "/" << epochs << std::endl;
      output.flush();
      newEpoch = false;
```

The functionality should be the same, so, your choice here. :+1:  The suggestion above would also allow us to keep `optimizer` commented out in the function definition.

Reply:
Hi, @rcurtin I incorporated these changes suggested by you in the previous commit by didn't check it for `MaxIterations==0` case. It was not showing the desired output then but I made the necessary changes in this commit and now it is working fine. Kindly have a look.